analytic_secondaxis/analytic_secondaxis.py

```diff
@@ -0,0 +1,291 @@
+# Service Company
+#
+# This program is Free Software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+#
+##############################################################################
+
+from openerp import models, fields, api
+import openerp.addons.decimal_precision as dp
+
+
+class ProjectActivityAl(models.Model):
+    _name = "project.activity_al"
+    _description = "Second Analytical Axes"
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        analytic_account_ids = []
+        if self._context.get('from_date', False):
+            args.append(['date', '>=', self._context['from_date']])
+        if self._context.get('to_date', False):
+            args.append(['date', '<=', self._context['to_date']])
+        if self._context.get('account_id', False):
+            analytic_account_obj = self.env['account.analytic.account']
+            analytic_account = analytic_account_obj.browse(
+                self._context.get('account_id'))
+            # take the account which have activity_ids
+            aa_activities = self._get_first_aa_which_have_activity(
+                analytic_account)
+            if aa_activities:
+                for temp_account in aa_activities:
+                    analytic_account_ids.append(temp_account.id)
+                args.append(('id', 'in', analytic_account_ids))
+        return super(ProjectActivityAl, self).search(
+            args,
+            offset=offset,
+            limit=limit,
+            order=order,
+            count=count
+        )
+
+    @api.multi
+    def _get_first_aa_which_have_activity(self, account):
+        """Return browse record list of activities of the account which
+        have an activity set (goes bottom up, child, then parent)
+        """
+        account.ensure_one()
+        if account.activity_ids:
+            return account
+        else:
+            if account.parent_id:
+                return self._get_first_aa_which_have_activity(
+                    account.parent_id
+                )
+            else:
+                return False
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=80):
+        """Override name_search function that perform the search
+        on the name of the activities
+        """
+        if not args:
```

Comment:
Maybe it's better to have a display_name and change rec_name to that one than performing this complicated searches.